olive-editor: fix build

ghpzin opened the pull request:
olive-editor: fix build of pinned openimageio

- add `override` to pinned `openimageio` with `fmt_10`, it does not
build with new default `fmt_11`

Fixes build failure of pinned `openimageio`:
```
CMake Error at src/cmake/modules/Findfmt.cmake:19 (string):
  string sub-command REGEX, mode MATCHALL needs at least 5 arguments total to
  command.
Call Stack (most recent call first):
  src/cmake/externalpackages.cmake:325 (find_package)
  src/cmake/externalpackages.cmake:354 (find_or_download_fmt)
  CMakeLists.txt:166 (include)
```

---

olive-editor: fix build with qt6-6.10

- add patch from merged upstream PR of `kddockwidgets` submodule:
https://www.github.com/KDAB/KDDockWidgets/pull/615
https://github.com/KDAB/KDDockWidgets/commit/f2b50fff29bd4b49acdfed3ed8fc42eb0a502032
(does not apply with `fetchpatch`)

Fixes build failure with qt6-6.10:
```
CMake Error at ext/KDDockWidgets/src/CMakeLists.txt:306 (target_link_libraries):
  Target "kddockwidgets" links to:

    Qt6::WidgetsPrivate

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.
```

---

olive-editor: cleanup

- replace `version` with correct format

---

Fixes #452183 - build failure of `olive-editor` since `2025-10-10` (logs of build failure of pinned `openimageio` inside):
https://hydra.nixos.org/build/310067751
https://cache.nixos.org/log/agrjbsihadkngj2w9qnixr8dml0ri96y-openimageio-2.4.15.0.drv
```text
CMake Error at src/cmake/modules/Findfmt.cmake:19 (string):
  string sub-command REGEX, mode MATCHALL needs at least 5 arguments total to
  command.
Call Stack (most recent call first):
  src/cmake/externalpackages.cmake:325 (find_package)
  src/cmake/externalpackages.cmake:354 (find_or_download_fmt)
  CMakeLists.txt:166 (include)
...
-- Configuring incomplete, errors occurred!
```

---


If you try to pass cmake variable quoting error with something like:
```diff
--- a/pkgs/by-name/ol/olive-editor/package.nix
+++ b/pkgs/by-name/ol/olive-editor/package.nix
@@ -30,6 +30,11 @@

     # robin-map headers require c++17
     cmakeFlags = (old.cmakeFlags or [ ]) ++ [ (lib.cmakeFeature "CMAKE_CXX_STANDARD" "17") ];
+
+    postPatch = ''
+      substituteInPlace src/cmake/modules/Findfmt.cmake \
+        --replace-fail 'REGEX MATCHALL "[0-9]+[.0-9]+" FMT_VERSION ''${TMP}' 'REGEX MATCHALL "[0-9]+[.0-9]+" FMT_VERSION "''${TMP}"'
+    '';
   });
 in
```

build will go further, but then fail with error not being able to find `fmt`:
```text
/build/source/src/libutil/ustring_test.cpp:21:14: fatal error: OpenImageIO/detail/fmt/std.h: No such file or directory
    21 | #    include <OpenImageIO/detail/fmt/std.h>
       |              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
make[2]: *** [src/libutil/CMakeFiles/ustring_test.dir/build.make:79: src/libutil/CMakeFiles/ustring_test.dir/ustring_test.cpp.o] Error 1
```
They seem to parse this `FMT_VERSION` from `include/fmt/core.h`:
https://github.com/AcademySoftwareFoundation/OpenImageIO/blob/9984e930004c6334cd33fe9c37a1324665c3ecc3/src/cmake/modules/Findfmt.cmake#L18
https://github.com/fmtlib/fmt/blob/e69e5f977d458f2650bb346dadf2ad30c5320281/include/fmt/core.h#L21
which does not exist in `fmt_11`:
https://github.com/fmtlib/fmt/blob/11.2.0/include/fmt/core.h
it is in `include/fmt/base.h` now:
https://github.com/fmtlib/fmt/blob/40626af88bd7df9a5fb80be7b25ac85b122d6c21/include/fmt/base.h#L24
We can patch new path instead, but that could change in newer `fmt` version and it does not seem like this `openimageio` pin is ever going to change.

Or backport:
https://github.com/AcademySoftwareFoundation/OpenImageIO/commit/65cfc19d06fc69551107bed08ca848928e6669a2
but that has more potential for issues.

<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform:
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [x] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [x] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [x] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [x] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

ghpzin committed:
olive-editor: fix build of pinned openimageio

- add `override` to pinned `openimageio` with `fmt_10`, it does not
build with new default `fmt_11`

Fixes build failure of pinned `openimageio`:
```
CMake Error at src/cmake/modules/Findfmt.cmake:19 (string):
  string sub-command REGEX, mode MATCHALL needs at least 5 arguments total to
  command.
Call Stack (most recent call first):
  src/cmake/externalpackages.cmake:325 (find_package)
  src/cmake/externalpackages.cmake:354 (find_or_download_fmt)
  CMakeLists.txt:166 (include)
```

maevii reviewed (changes requested):
Still fails to build on my x86_64-linux, here are the full logs:

<details>
<summary>1qm0pxphfp70ss1c8l8xq11fjm347ip-olive-editor-unstable-2023-06-12.drv.log</summary>

```
Running phase: qtPreHook
@nix { "action": "setPhase", "phase": "qtPreHook" }
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/zpig3vzwq92ghmzs1lg462rcswz5vq86-source
source root is source
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
applying patch /nix/store/n3p7g42w767bf9zvcdmw3m5v0agswm8s-olive-editor-openimageio-2.3-compat.patch
patching file app/render/renderer.cpp
substituteStream() in derivation olive-editor-unstable-2023-06-12: WARNING: '--replace' is deprecated, use --replace-{fail,warn,quiet}. (file './app/node/project/serializer/serializer230220.cpp')
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
cmake flags: -DCMAKE_FIND_USE_SYSTEM_PACKAGE_REGISTRY=OFF -DCMAKE_FIND_USE_PACKAGE_REGISTRY=OFF -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_BUILD_TYPE=Release -DBUILD_TESTING=OFF -DCMAKE_INSTALL_LOCALEDIR=/nix/store/ngr6b8ik3vxam4zdj40y6y5fdjj7h6wf-olive-editor-unstable-2023-06-12/share/locale -DCMAKE_INSTALL_LIBEXECDIR=/nix/store/ngr6b8ik3vxam4zdj40y6y5fdjj7h6wf-olive-editor-unstable-2023-06-12/libexec -DCMAKE_INSTALL_LIBDIR=/nix/store/ngr6b8ik3vxam4zdj40y6y5fdjj7h6wf-olive-editor-unstable-2023-06-12/lib -DCMAKE_INSTALL_DOCDIR=/nix/store/ngr6b8ik3vxam4zdj40y6y5fdjj7h6wf-olive-editor-unstable-2023-06-12/share/doc/olive-editor -DCMAKE_INSTALL_INFODIR=/nix/store/ngr6b8ik3vxam4zdj40y6y5fdjj7h6wf-olive-editor-unstable-2023-06-12/share/info -DCMAKE_INSTALL_MANDIR=/nix/store/ngr6b8ik3vxam4zdj40y6y5fdjj7h6wf-olive-editor-unstable-2023-06-12/share/man -DCMAKE_INSTALL_INCLUDEDIR=/nix/store/ngr6b8ik3vxam4zdj40y6y5fdjj7h6wf-olive-editor-unstable-2023-06-12/include -DCMAKE_INSTALL_SBINDIR=/nix/store/ngr6b8ik3vxam4zdj40y6y5fdjj7h6wf-olive-editor-unstable-2023-06-12/sbin -DCMAKE_INSTALL_BINDIR=/nix/store/ngr6b8ik3vxam4zdj40y6y5fdjj7h6wf-olive-editor-unstable-2023-06-12/bin -DCMAKE_INSTALL_NAME_DIR=/nix/store/ngr6b8ik3vxam4zdj40y6y5fdjj7h6wf-olive-editor-unstable-2023-06-12/lib -DCMAKE_POLICY_DEFAULT_CMP0025=NEW -DCMAKE_FIND_FRAMEWORK=LAST -DCMAKE_STRIP=/nix/store/x8mydcgbry214s802nzvy7fdljx404ym-gcc-wrapper-14.3.0/bin/strip -DCMAKE_RANLIB=/nix/store/x8mydcgbry214s802nzvy7fdljx404ym-gcc-wrapper-14.3.0/bin/ranlib -DCMAKE_AR=/nix/store/x8mydcgbry214s802nzvy7fdljx404ym-gcc-wrapper-14.3.0/bin/ar -DCMAKE_C_COMPILER=gcc -DCMAKE_CXX_COMPILER=g++ -DCMAKE_INSTALL_PREFIX=/nix/store/ngr6b8ik3vxam4zdj40y6y5fdjj7h6wf-olive-editor-unstable-2023-06-12 -DBUILD_QT6=1 -DQT_OPTIONAL_TOOLS_PATH=/nix/store/qld4r900hwnf2qq2pl6v1811zgpj4i51-qttools-6.10.0 -DQT_OPTIONAL_TOOLS_PATH=/nix/store/qld4r900hwnf2qq2pl6v1811zgpj4i51-qttools-6.10.0
-- The CXX compiler identification is GNU 14.3.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /nix/store/x8mydcgbry214s802nzvy7fdljx404ym-gcc-wrapper-14.3.0/bin/g++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found OpenGL: /nix/store/as2rg79vk4z7frb7hy27nmadc3fp1a4j-libglvnd-1.7.0/lib/libOpenGL.so
-- Found OpenColorIO: /nix/store/6v2alwkjl099bfpii8hhgdcxnxpv2vrv-opencolorio-2.4.2/lib/libOpenColorIO.so (found suitable version "2.4.2", minimum required is "2.1.1")
-- Found OpenImageIO: /nix/store/4qkfzzc9yzhvbckj1cz84nn0bhkh7js7-openimageio-2.4.15.0/lib/libOpenImageIO.so;/nix/store/4qkfzzc9yzhvbckj1cz84nn0bhkh7js7-openimageio-2.4.15.0/lib/libOpenImageIO_Util.so (found suitable version "2.4.15", minimum required is "2.1.12")
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE
-- Found CONFIG for OpenEXR 3 (OPENEXR_VERSION=3.3.5)
-- Performing Test HAVE_STDATOMIC
-- Performing Test HAVE_STDATOMIC - Success
-- Found WrapAtomic: TRUE
-- Found WrapOpenGL: TRUE
-- Found WrapVulkanHeaders: /nix/store/icyb0ffm7i4sajzphnf818firkqlx71h-vulkan-headers-1.4.328.0/include
-- Found FFMPEG: /nix/store/yjanmflpijsapfrv9l371kayrdzx6gw2-ffmpeg-6.1.2-dev/include (found suitable version "6.1.2", minimum required is "3.0") found components: avutil avcodec avformat avfilter swscale swresample
-- Found PortAudio: /nix/store/mz9n0p4x390ypqwm3055783m5c8lp061-portaudio-190700_20210406/lib/libportaudio.so
-- Looking for C++ include pa_jack.h
-- Looking for C++ include pa_jack.h - found
-- Could NOT find OpenTimelineIO (missing: OTIO_LIBRARIES OTIO_INCLUDE_DIRS OTIO_DEPS_INCLUDE_DIR) 
   OpenTimelineIO interchange will be disabled.
-- Could NOT find GoogleCrashpad (missing: CRASHPAD_CLIENT_LIB CRASHPAD_COMMON_LIB CRASHPAD_UTIL_LIB CRASHPAD_BASE_LIB BREAKPAD_BIN_DIR CRASHPAD_BUILD_INCLUDE_DIR CRASHPAD_CLIENT_INCLUDE_DIR CRASHPAD_BASE_INCLUDE_DIR CRASHPAD_COMPAT_LIB) 
   Automatic crash reporting will be disabled.
-- Found FFMPEG: /nix/store/yjanmflpijsapfrv9l371kayrdzx6gw2-ffmpeg-6.1.2-dev/include (found suitable version "6.1.2", minimum required is "3.0") found components: avutil
-- Could NOT find OpenTimelineIO (missing: OTIO_LIBRARIES OTIO_INCLUDE_DIRS OTIO_DEPS_INCLUDE_DIR) 
   OpenTimelineIO interchange will be disabled.
-- The following OPTIONAL packages have been found:

 * Qt6Core
 * Vulkan
 * Qt6DBusTools (required version >= 6.10.0)
 * Qt6Gui
 * Qt6CoreTools (required version >= 6.10.0)
 * Qt6GuiTools (required version >= 6.10.0)
 * Qt6WidgetsTools (required version >= 6.10.0)
 * Qt6Widgets
 * WrapVulkanHeaders
 * Qt6OpenGL
 * Qt6LinguistTools
 * Qt6Concurrent
 * Qt6Network
 * Qt6OpenGLWidgets
 * Qt6Test (required version >= 6.2.0)

-- The following REQUIRED packages have been found:

 * OpenColorIO (required version >= 2.1.1)
 * OpenImageIO (required version >= 2.1.12)
 * OpenEXR
 * Threads
 * QT
 * PortAudio
 * FFMPEG (required version >= 3.0)
 * Imath
 * OpenGL
 * Qt6 (required version >= 6.2.0)

-- The following OPTIONAL packages have not been found:

 * GoogleCrashpad
 * OpenTimelineIO

-- Configuring done (2.4s)
CMake Error at ext/KDDockWidgets/src/CMakeLists.txt:306 (target_link_libraries):
  Target "kddockwidgets" links to:

    Qt6::WidgetsPrivate

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.



-- Generating done (0.3s)
CMake Warning:
  Manually-specified variables were not used by the project:

    BUILD_TESTING
    CMAKE_C_COMPILER
    CMAKE_EXPORT_NO_PACKAGE_REGISTRY
    CMAKE_POLICY_DEFAULT_CMP0025


CMake Generate step failed.  Build files cannot be regenerated correctly.
```
</details>

ghpzin commented:
New issue is related to `qt6` bump to `6.10` in:
https://github.com/NixOS/nixpkgs/pull/449572
https://bugreports.qt.io/browse/QTBUG-87776
will try to add fix for that too.

ghpzin committed:
olive-editor: fix build with qt6-6.10

- add patch from merged upstream PR of `kddockwidgets` submodule:
https://www.github.com/KDAB/KDDockWidgets/pull/615
https://github.com/KDAB/KDDockWidgets/commit/f2b50fff29bd4b49acdfed3ed8fc42eb0a502032
(does not apply with `fetchpatch`)

Fixes build failure with qt6-6.10:
```
CMake Error at ext/KDDockWidgets/src/CMakeLists.txt:306 (target_link_libraries):
  Target "kddockwidgets" links to:

    Qt6::WidgetsPrivate

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.
```

maevii reviewed (approved):
Builds & runs fine now. I'm nowhere near familiar enough with the package and Qt to say whether this is the right way to fix it, but it does work and looks fine from a distance.

ghpzin committed:
olive-editor: cleanup

- replace `version` with correct format
